### DISCUSSION TITLE
Support for arbitrary build settings

### DISCUSSION THREAD

**Discussion by kamcma** [mxcl/xcodebuild#83] [Ideas]
Per its man page, `xcodebuild` can accept arbitrary build settings:
```bash
xcodebuild [action ...] [buildsetting=value ...]
```
This could be useful for any number of reasons, but my use case in CI is to derive version numbers from git, pass them to `xcodebuild` as settings, where they will be expanded into the build artifact's Info.plist file because the further build setting `INFOPLIST_EXPAND_BUILD_SETTINGS=YES`. Eg:
```bash
xcodebuild archive CURRENT_PROJECT_VERSION=8 MARKETING_VERSION=1.0.4 # actual numbers derived from git in a previous job step
```

If it were possible for this action to pass through arbitrary build settings to `xcodebuild`, I would certainly be interested in that functionality. Thanks for considering.

**Comment by mxcl** [comment]
Yeah this would be fine to add. And relatively easy. Probs 4 or 5 lines of code.
